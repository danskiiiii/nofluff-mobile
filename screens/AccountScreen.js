import { Image, KeyboardAvoidingView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import LoginForm from '../components/LoginForm';
import React from 'react';
import RegisterForm from '../components/RegisterForm';
import { connect } from 'react-redux';

class AccountScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };
  constructor(props) {
    super(props);
    this.state = { register: false };
  }
  render() {
    const { logged } = this.props;

    return (
      <KeyboardAvoidingView behavior="padding" style={styles.container}>
        <View style={styles.imageContainer}>
          <Image resizeMode="contain" style={styles.logo} source={require('../assets/images/logo.png')} />
        </View>
        <TouchableOpacity onPress={() => this.setState({ register: !this.state.register })}>
          <Text style={styles.noAccountText}>{this.state.register ? 'Back to login.' : 'No account? Tap here.'}</Text>
        </TouchableOpacity>
        <View>
          {!logged && !this.state.register && <LoginForm />}
          {this.state.register && <RegisterForm />}
        </View>
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  imageContainer: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center',
  },
  logo: {
    position: 'absolute',
    width: 300,
    height: 100,
  },
  noAccountText: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '300',
    backgroundColor: '#d8d8d8',
    marginLeft: 20,
    marginRight: 20,
  },
});

function mapStateToProps(state) {
  return {
    logged: state.auth.isLoggedIn,
  };
}

export default connect(mapStateToProps)(AccountScreen);
