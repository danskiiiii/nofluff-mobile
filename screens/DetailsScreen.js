import { ActivityIndicator, Image, StyleSheet, View } from 'react-native';

import OfferDetail from '../components/details/OfferDetail';
import OpinionDetail from '../components/details/OpinionDetail';
import React from 'react';
import { connect } from 'react-redux';

class DetailsScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    const { isLoading, navigation } = this.props;
    const data = navigation.getParam('data') && JSON.parse(navigation.getParam('data'));
    const type = navigation.getParam('type');
    return (
      <View style={styles.container}>
        {type && (
          <View style={styles.imageContainer}>
            <Image resizeMode="contain" style={styles.logo} source={require('../assets/images/logo.png')} />
          </View>
        )}

        {isLoading && (
          <View style={styles.bigImageContainer}>
            <ActivityIndicator style={styles.bigLogo} size="large" color="#2980b6" />
          </View>
        )}

        {!isLoading && type === 'offer' && <OfferDetail data={data} />}

        {!isLoading && type === 'opinion' && <OpinionDetail data={data} />}

        {!isLoading &&
          type === undefined && (
            <View style={styles.bigImageContainer}>
              <Image resizeMode="contain" style={styles.bigLogo} source={require('../assets/images/logo.png')} />
            </View>
          )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  imageContainer: {
    alignItems: 'center',
    height: 60,
    borderBottomWidth: 1,
    borderColor: '#c1c1c1',
  },
  logo: {
    bottom: 0,
    width: 150,
    height: 80,
  },
  bigImageContainer: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center',
  },
  bigLogo: {
    position: 'absolute',
    width: 300,
    height: 100,
  },
  spinner: {
    position: 'absolute',
    width: 100,
    height: 100,
  },
});

function mapStateToProps(state) {
  return {
    isLoading: state.ratings.loading,
  };
}

export default connect(mapStateToProps)(DetailsScreen);
