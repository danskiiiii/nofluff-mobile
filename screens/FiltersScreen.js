import { Picker, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';

import React from 'react';
import { getOffers } from '../actions/creators/offers';
import store from '../store';

class FiltersScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  constructor(props) {
    super(props);
    this.state = { location: '', experience: '' };
  }

  onFilterPress = () => {
    store
      .dispatch(getOffers(this.state.location, this.state.experience))
      .then(() => this.props.navigation.navigate('Offers'));
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.borderContainer}>
          <TextInput
            placeholder="Location"
            style={styles.input}
            autoCapitalize="words"
            autoCorrect={false}
            keyboardType="default"
            returnKeyType="next"
            placeholderTextColor="rgba(0,0,0,0.35)"
            value={this.state.email}
            onChangeText={location => this.setState({ location })}
          />
        </View>
        <View style={styles.borderContainer}>
          <Text style={styles.labelText}>Required experience</Text>
          <Picker selectedValue={this.state.experience} onValueChange={experience => this.setState({ experience })}>
            <Picker.Item label="Any" value="" />
            <Picker.Item label="Junior" value="junior" />
            <Picker.Item label="Regular" value="regular" />
            <Picker.Item label="Senior" value="senior" />
          </Picker>
        </View>
        <TouchableOpacity style={styles.buttonContainer} onPress={this.onFilterPress}>
          <Text style={styles.buttonText}>Filter</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffff',
    padding: 20,
    justifyContent: 'center',
  },
  input: {
    height: 40,
    marginBottom: 10,
    padding: 10,
    color: '#000',
  },
  borderContainer: {
    borderWidth: 2,
    borderRadius: 3,
    borderColor: '#2980b6',
    marginBottom: 5,
  },
  labelText: {
    marginLeft: 7,
    marginRight: 7,
    marginTop: 10,
    fontSize: 16,
    textDecorationLine: 'underline',
  },
  buttonContainer: {
    backgroundColor: '#2980b6',
    paddingVertical: 10,
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
    fontWeight: '400',
  },
});
export default FiltersScreen;
