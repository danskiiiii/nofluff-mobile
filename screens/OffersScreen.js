import { Alert, Image, ScrollView, StyleSheet, View } from 'react-native';

import OfferPreviewCard from '../components/OfferPreviewCard';
import React from 'react';
import { WebBrowser } from 'expo';
import { connect } from 'react-redux';
import { getOffers } from '../actions/creators/offers';

class OffersScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  componentDidMount() {
    this.props.dispatch(getOffers());
  }

  handleClickTest = () => {
    WebBrowser.openBrowserAsync(`https://www.google.com/search?q=${this.props.data[1].username}`);
    Alert.alert(JSON.stringify(this.props.data));
  };

  renderOffers = () => {
    const { navigate } = this.props.navigation;
    const { data } = this.props;
    return data.map(element => (
      <OfferPreviewCard
        key={element.id}
        text1={element.name}
        text2={element.username}
        onPress={() =>
          navigate('Details', {
            offerId: element.id,
            data: JSON.stringify(element),
          })
        }
      />
    ));
  };

  render() {
    const { navigate } = this.props.navigation;

    return (
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image resizeMode="contain" style={styles.logo} source={require('../assets/images/logo.png')} />
        </View>
        <ScrollView style={styles.scrollContainer}>
          <OfferPreviewCard onPress={this.handleClickTest} text="Co" />
          <OfferPreviewCard
            onPress={() =>
              navigate('Details', {
                itemId: 86,
                otherParam: 'anything you want here',
              })
            }
            text="to"
          />
          <OfferPreviewCard onPress={() => navigate('Opinions')} text="za" />
          <OfferPreviewCard onPress={() => navigate('Account')} text="apka" />
          {this.renderOffers()}
        </ScrollView>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#c1c1c1',
  },
  imageContainer: {
    alignItems: 'center',
    backgroundColor: '#ffffff',
    height: 60,
    borderBottomWidth: 1,
    borderColor: '#c1c1c1',
  },
  logo: {
    bottom: 0,
    width: 150,
    height: 80,
  },

  scrollContainer: {
    flex: 1,
    position: 'relative',
    marginBottom: 6,
    backgroundColor: '#c1c1c1',
  },
});

function mapStateToProps(state) {
  return {
    data: state.offers.data,
  };
}

export default connect(mapStateToProps)(OffersScreen);
