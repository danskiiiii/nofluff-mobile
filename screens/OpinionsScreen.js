import { Alert, Image, ScrollView, StyleSheet, View } from 'react-native';
import { getCompanies, getOpinions } from '../actions/creators/ratings';

import OpinionPreviewCard from '../components/OpinionPreviewCard';
import React from 'react';
// import { WebBrowser } from 'expo';
import { connect } from 'react-redux';

class OpinionsScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  componentDidMount() {
    this.props.dispatch(getCompanies());
  }

  handleClickTest = () => {
    // WebBrowser.openBrowserAsync(`https://www.google.com/search?q=${this.props}`);
    Alert.alert(JSON.stringify(this.props.data));
  };

  renderCompanies = () => {
    const { navigate } = this.props.navigation;
    const { data, dispatch } = this.props;
    return data.map(element => (
      <OpinionPreviewCard
        key={element.id}
        company={element.company_name}
        location={element.location}
        logoUri={element.logo}
        rating={element.rating}
        onPress={() =>
          dispatch(getOpinions(element.id)).then(
            navigate('Details', { data: JSON.stringify(element), type: 'opinion' })
          )
        }
      />
    ));
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image resizeMode="contain" style={styles.logo} source={require('../assets/images/logo.png')} />
        </View>
        <ScrollView style={styles.scrollContainer}>{this.renderCompanies()}</ScrollView>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#c1c1c1',
  },
  imageContainer: {
    alignItems: 'center',
    backgroundColor: '#ffffff',
    height: 60,
    borderBottomWidth: 1,
    borderColor: '#c1c1c1',
  },
  logo: {
    bottom: 0,
    width: 150,
    height: 80,
  },

  scrollContainer: {
    flex: 1,
    position: 'relative',
    marginBottom: 6,
    backgroundColor: '#c1c1c1',
  },
});

function mapStateToProps(state) {
  return {
    data: state.ratings.data,
  };
}

export default connect(mapStateToProps)(OpinionsScreen);
