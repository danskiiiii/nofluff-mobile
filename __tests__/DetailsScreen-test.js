import 'react-native';
import 'jsdom-global/register';

import { mount, shallow } from 'enzyme';

import { DetailsScreen } from '../screens/DetailsScreen';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import React from 'react';
import renderer from 'react-test-renderer';

const defaultNavData = {
  getParam: jest.fn(),
};
const opinionNavData = {
  getParam: param => (param === 'data' ? '' : 'opinion'),
};
const offerNavData = {
  getParam: param => (param === 'data' ? '' : 'offer'),
};

describe('DetailsScreen snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the DetailsScreen', async () => {
    const tree = renderer.create(<DetailsScreen navigation={defaultNavData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

function getComponent(navigation, isLoading) {
  return <DetailsScreen navigation={navigation} isLoading={isLoading} />;
}

describe('DetailsScreen component', () => {
  it('renders', () => {
    const wrapper = shallow(getComponent(defaultNavData));

    expect(wrapper.exists()).toBe(true);
  });

  it('renders Image', () => {
    const wrapper = mount(getComponent(defaultNavData));

    expect(wrapper.find('Image').exists()).toBe(true);
  });

  it('renders spinner when awaiting data', () => {
    const wrapper = mount(getComponent(defaultNavData, true));

    expect(wrapper.find('ActivityIndicator').exists()).toBe(true);
  });

  it('renders OpinionDetail', () => {
    const wrapper = shallow(getComponent(opinionNavData, false));

    expect(wrapper.find('withNavigation(Connect(OpinionDetail))').exists()).toBe(true);
  });

  it('renders OfferDetail', () => {
    const wrapper = shallow(getComponent(offerNavData, false));

    expect(wrapper.find('withNavigation(Connect(OfferDetail))').exists()).toBe(true);
  });
});
