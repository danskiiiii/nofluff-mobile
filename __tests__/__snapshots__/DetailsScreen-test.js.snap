// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen snapshot renders the DetailsScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 100,
          "position": "absolute",
          "width": 300,
        }
      }
    />
  </View>
</View>
`;
